core/ang/afformCore.js

```diff
@@ -3,19 +3,28 @@
   angular.module('afformCore', CRM.angRequires('afformCore'));
 
   // Use `afformCoreDirective(string name)` to generate an AngularJS directive.
-  angular.module('afformCore').service('afformCoreDirective', function($routeParams, crmApi4, crmStatus, crmUiAlert){
-    return function(camelName, meta, d){
+  angular.module('afformCore').service('afformCoreDirective', function(crmApi4, crmStatus, crmUiAlert) {
+    return function(camelName, meta, d) {
+      function getUrlVars() {
+        var vars = {};
+        window.location.search.replace(/[?&]+([^=&]+)=([^&]*)/gi, function(m, key, value) {
+          vars[key] = value;
+        });
+        return vars;
+      }
       d.restrict = 'AE';
       d.scope = {};
       d.scope.options = '=' + camelName;
-      d.link = function($scope, $el, $attr) {
-        $scope.ts = CRM.ts(camelName);
-        $scope.routeParams = $routeParams;
-        $scope.meta = meta;
-        $scope.crmApi4 = crmApi4;
-        $scope.crmStatus = crmStatus;
-        $scope.crmUiAlert = crmUiAlert;
-        $scope.crmUrl = CRM.url;
+      d.link = {
+        pre: function($scope, $el, $attr) {
+          $scope.ts = CRM.ts(camelName);
+          $scope.routeParams = getUrlVars();
```

Comment:
So this is the substitute for ngRoute since so many examples seem to depend on it.